presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardPredicate.java

```diff
@@ -138,8 +138,20 @@ public static ShardPredicate create(TupleDomain<RaptorColumnHandle> tupleDomain)
                 }
             }
 
-            String min = minColumn(handle.getColumnId());
-            String max = maxColumn(handle.getColumnId());
+            String min;
+            String max;
+            if (handle.isBucketNumber()) {
+                if (!bucketed) {
+                    predicate.add("true");
```

Comment:
Per our discussion, it seems this can be `false`. Please add a test which would hit this case.